### PR TITLE
Integrate table select with pagination/search

### PR DESCRIPTION
**Pull request by gabibguti**
### Description
Use `compareElements` function to decide if this element was previously selected or not when changing pages or searching.

> New props - ListTableTemplate / Table
```javascript
compareElements: PropTypes.func
```

Also added a prop to set initially `checked` rows on table.

> New props - Data / Table
```javascript
/** List of elements index to start selected. */
startSelected: PropTypes.array,
```

When using select you should set a `compareElements` customized function.
Example:
```javascript
<ListTableTemplate.Table
    ...
    selectable
    compareElements = (curr, next) => curr.id === next.id
/>
```

### Testing
https://dev--gc-lea7269.mygocommerce.com/admin/mercadolivre/products/
```javascript
console.log('SELECTED:', selectedProducts)
```

### CODE REVIEW

File: react/advancedComponents/ListTableTemplate/components/Table/index.js [hunk 1]
```diff
@@ -1,14 +1,53 @@
...
 
 class Table extends PureComponent {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      updateTable: 0,
+      rowsData: [],
+      cacheData: [],
```
Comment (augustolazaro) [inline react/advancedComponents/ListTableTemplate/components/Table/index.js]: Maybe we don't need this array and use only the list of selected items. When a new page is loaded you can check if each item is on the selected list then

File: react/advancedComponents/ListTableTemplate/components/Table/index.js [hunk 2]
```diff
@@ -176,6 +243,8 @@ Table.propTypes = {
 
 Table.defaultProps = {
   onChange: () => { },
+  compareElements: (curr, next) => curr === next,
```
Comment (augustolazaro) [inline react/advancedComponents/ListTableTemplate/components/Table/index.js]:
```
compareElements: (curr, next) => JSON.stringify(curr) === JSON.stringify(next)
```

**Comment by augustolazaro** [comment]
Just be sure that the `react-intl` new major is not breaking anything

**Comment by gabibguti** [comment]
> Just be sure that the `react-intl` new major is not breaking anything

@augustolazaro 
It's not giving me any compiling errors so far, but I'm gonna check the usage in the code.

**Pull request merged**